src/main/java/uk/gov/hmcts/reform/em/hrs/service/email/AuditReportCsvWriter.java

```diff
@@ -0,0 +1,98 @@
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Collections.emptyList;
+
+@Component
+public class AuditReportCsvWriter {
+
+    private static final String[] HEARING_AUDIT_SUMMARY_CSV_HEADERS = {
+        "Action", "UserName", "File Name", "Source URI", "Hearing Source",
+        "Service Code", "File Size KB", "CCD Case Id", "Date Processed", "Shared On", "Sharee Email"
+    };
+
+    public File writeHearingRecordingSummaryToCsv(List<AuditEntry> data) throws IOException {
+        File csvFile = File.createTempFile(
+            "hearing-audit-report",
+            ".csv"
+        );
+
+        CSVFormat csvFileHeader = CSVFormat
+            .Builder
+            .create()
+            .setHeader(HEARING_AUDIT_SUMMARY_CSV_HEADERS)
+            .build();
+        try (
+            FileWriter fileWriter = new FileWriter(csvFile);
+            CSVPrinter printer = new CSVPrinter(fileWriter, csvFileHeader)
+        ) {
+            for (AuditEntry entry : Optional.ofNullable(data).orElse(emptyList())) {
+                if (entry instanceof HearingRecordingSegmentAuditEntry hearingAudit) {
+                    hearingAudit = (HearingRecordingSegmentAuditEntry) entry;
+                    var hearingRecSeg = hearingAudit.getHearingRecordingSegment();
+                    printer.printRecord(
+                        entry.getAction(),
+                        entry.getUsername(),
+                        hearingRecSeg.getFilename(),
+                        hearingRecSeg.getIngestionFileSourceUri(),
+                        hearingRecSeg.getHearingRecording().getHearingSource(),
+                        hearingRecSeg.getHearingRecording().getServiceCode(),
+                        (int) Math.ceil((float) hearingRecSeg.getFileSizeMb() / 1000),
+                        hearingRecSeg.getHearingRecording().getCcdCaseId(),
+                        hearingRecSeg.getCreatedOn(),
+                        "",
+                        ""
+                    );
+                } else if (entry instanceof HearingRecordingAuditEntry hearingRecordingAudit) {
+                    hearingRecordingAudit = (HearingRecordingAuditEntry) entry;
+                    var hearingRec = hearingRecordingAudit.getHearingRecording();
+                    printer.printRecord(
+                        entry.getAction(),
+                        entry.getUsername(),
+                        "",
+                        "",
+                        hearingRec.getHearingSource(),
+                        hearingRec.getServiceCode(),
+                        "",
+                        hearingRec.getCcdCaseId(),
+                        hearingRec.getCreatedOn(),
+                        "",
+                        ""
+                    );
+                } else if (entry instanceof HearingRecordingShareeAuditEntry hearingShareeAudit) {
+                    hearingShareeAudit = (HearingRecordingShareeAuditEntry) entry;
```

Comment:
We no more need this casting. Please remove this line.

Reply:
removed